WeakAuras/Prototypes.lua

```diff
@@ -1126,20 +1126,35 @@ local function AddUnitChangeInternalEvents(unit, t)
   end
 end
 
+local function AddUnitEventForEvents(unit, result, event)
+  if not WeakAuras.baseUnitId[unit] then
+    tinsert(result, event)
+  end
+end
+
+local function AddUnitEventForUnit_events(unit, result, event)
+  if WeakAuras.baseUnitId[unit] then
+    result[unit] = result[unit] or {}
+    tinsert(result[unit], event)
+  end
+end
+
```

Comment:
These 2 functions are helpers for populating "events" or "unit_events" depending if the unit provided is listed in WeakAuras.baseUnitId from Types.lua